src/components/app.jsx

```diff
@@ -1,18 +1,121 @@
 import React from 'react';
+import enigma from 'enigma.js';
+import SVGInline from 'react-svg-inline';
 
+import config from '../enigma/config';
 import Selections from './selections';
 import Model from './model';
-import { AppProvider } from './app-provider';
+import logo from '../assets/catwalk.svg';
 
 import './app.css';
 
-export const App = () => (
-  <AppProvider>
-    <div className="app">
-      <Selections />
-      <Model />
-    </div>
-  </AppProvider>
-);
+export const AppContext = React.createContext(null);
+export const AppConsumer = AppContext.Consumer;
 
-export default App;
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      app: null,
+      docs: null,
+      error: null,
+      engineURL: new URLSearchParams(document.location.search).get('engine_url') || 'ws://localhost:9076/',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  async componentDidMount() {
+    let qixGlobal;
+    const session = enigma.create(config);
+    try {
+      qixGlobal = await session.open();
+      const appHandle = await qixGlobal.getDoc(); // Mixin from ./src/enigma/get-doc
+      const appLayout = await appHandle.getAppLayout();
+      appHandle.lastReloadTime = appLayout.qLastReloadTime;
+      this.setState({ session, app: appHandle });
+    } catch (error) {
+      if (qixGlobal) {
+        const docs = await qixGlobal.getDocList();
+        this.setState({ docs });
+      } else {
+        this.setState({ error });
+      }
+    }
+  }
+
+  componentWillUnmount() {
+    const { session } = this.state;
+    if (session) {
+      session.close();
+    }
+  }
+
+  handleSubmit(event, docId) {
+    let { engineURL } = this.state;
+    if (docId) {
+      engineURL = `${new URL(engineURL).origin}${docId}`;
+    } else {
+      event.preventDefault();
+    }
+    window.history.replaceState({}, '', `${window.location.pathname}?engine_url=${encodeURI(engineURL)}`);
+    window.location.reload(false);
+  }
+
+  handleChange(event) {
+    this.setState({ engineURL: event.target.value });
+  }
+
+  render() {
+    const {
+      app,
+      docs,
+      error,
+      engineURL,
+    } = this.state;
+
+    // Render the "HUB" (global is defined == connection to engine)
```

Comment:
The forbidden word... 🙀

Reply:
Missed it :) - will remove it then creating component of the **HUB**